### PR TITLE
#652: Started unit tests for script option lines parser

### PR DESCRIPTION
**Pull request by tomuben**
related to https://github.com/exasol/script-languages-release/issues/652

### CODE REVIEW

File: exaudfclient/base/exaudflib/test/script_data_transfer_objects_test.cpp [hunk 1]
```diff
@@ -258,9 +258,3 @@ TEST_F(ConnectionInformationTest, copy)
     EXPECT_TRUE(ca.getPassword() == cb.getPassword());
     EXPECT_TRUE(ca.hasData() == cb.hasData());
 }
-
-
-int main(int argc, char **argv) {
-    ::testing::InitGoogleTest(&argc, argv);
-    return RUN_ALL_TESTS();
-}
```
Comment (tomuben) [inline exaudfclient/base/exaudflib/test/script_data_transfer_objects_test.cpp:266]: Covered now by Bazel modul, "@googletest//:gtest_main"

File: exaudfclient/base/exaudflib/test/script_data_transfer_objects_test.cpp [hunk 2]
```diff
@@ -258,9 +258,3 @@ TEST_F(ConnectionInformationTest, copy)
     EXPECT_TRUE(ca.getPassword() == cb.getPassword());
     EXPECT_TRUE(ca.hasData() == cb.hasData());
```
Comment (Nicoretti) [inline exaudfclient/base/exaudflib/test/script_data_transfer_objects_test.cpp:259]: I know this did not change, but this looks really wrong to me! `ASSERT_TRUE(true)`

Likely something along the lines like this:
```cpp
    EXPECT_TRUE(ca.hasData());
    EXPECT_STREQ(ca.getPassword(), "password" );
```

would make more sense.

File: exaudfclient/base/exaudflib/test/script_option_lines_test.cpp [hunk 3]
```diff
@@ -0,0 +1,172 @@
...
+    size_t pos;
+    std::string code =
+        "%option myoption; %option mysecondoption;\n"
+        "\nmycode";
+    const std::string res = extractOptionLine(code, "%option", whitespace, lineEnd, pos, throwException);
+    EXPECT_EQ(res, "myoption");
+    const std::string expected_resulting_code =
+        " %option mysecondoption;\n"
+        "\nmycode";
+
```
Comment (Nicoretti) [inline exaudfclient/base/exaudflib/test/script_option_lines_test.cpp:80]: If there are no modifications or side effects I think test read better when the expectations are asserted at the end.

AAA = Arange Act Assert
```suggestion
    const std::string expected_resulting_code =
        " %option mysecondoption;\n"
        "\nmycode";
    EXPECT_EQ(res, "myoption");
```

File: exaudfclient/base/exaudflib/test/script_option_lines_test.cpp [hunk 4]
```diff
@@ -0,0 +1,172 @@
...
+    size_t pos;
+    std::string code =
+        "%option myoption\n"
+        "\nmycode";
+   EXPECT_THROW({
+        const std::string res = extractOptionLine(code, "%option", whitespace, lineEnd, pos, throwException);
+    }, TestException );
+}
+
+TEST(ScriptOptionLinesTest, only_finds_the_first_option) {
```
Comment (tkilias) [inline exaudfclient/base/exaudflib/test/script_option_lines_test.cpp:70]: We also need a test with multiple different options.

Comment (tomuben) [reply exaudfclient/base/exaudflib/test/script_option_lines_test.cpp:70]: Good point. Added it.

File: exaudfclient/base/exaudflib/test/script_option_lines_test.cpp (hunk 3 above)
Comment (tomuben) [reply exaudfclient/base/exaudflib/test/script_option_lines_test.cpp:80]: I agree. Changed the tests.

File: exaudfclient/base/exaudflib/test/script_data_transfer_objects_test.cpp (hunk 2 above)
Comment (tomuben) [reply exaudfclient/base/exaudflib/test/script_data_transfer_objects_test.cpp]: Done

**Review by Nicoretti — COMMENTED** [review]
Do the tests work because AFIK `EXPECT_EQ` compares pointers not content [EXPECT_EQ](https://github.com/google/googletest/blob/main/docs/reference/assertions.md#expect_eq-expect_eq) vs [EXPECT_STREQ](https://github.com/google/googletest/blob/main/docs/reference/assertions.md#expect_streq-expect_streq)

Alternatively GMock matcher style is also possible:

`EXPECT_THAT(my_str, StrEq("expected"));`

**Comment by tomuben** [comment]
> Do the tests work because AFIK `EXPECT_EQ` compares pointers not content [EXPECT_EQ](https://github.com/google/googletest/blob/main/docs/reference/assertions.md#expect_eq-expect_eq) vs [EXPECT_STREQ](https://github.com/google/googletest/blob/main/docs/reference/assertions.md#expect_streq-expect_streq)
> 
> Alternatively GMock matcher style is also possible:
> 
> `EXPECT_THAT(my_str, StrEq("expected"));`

The values on the left side are of type `std::string`. That's the reason it works.....

**Review by Nicoretti — APPROVED** [review]

**Review by tkilias — APPROVED** [review]

**Pull request merged**